### PR TITLE
Mass Module: support multiple custom configs for dealId and rendering

### PR DESCRIPTION
**Pull request by cciocov**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
Updated the MASS module to support multiple parallel dealId <-> renderer configurations.

### CODE REVIEW

**Comment by patmmccann** [comment]
fyi @Fawke this is the followup commit from https://github.com/prebid/Prebid.js/pull/6332 ; the discussion there you may find interesting. The idea is that this module can be adapted to link renderers to deal ids if this use case arises in the future without each company looking for this functionality requiring a deal rendering module.

**Comment by patmmccann** [comment]
Tagging as needing two reviews because this commit converts a vendor specific module into functionality which is generic to many use cases

**Comment by patmmccann** [comment]
Hi @cciocov your docs are not linked from the js modules directory https://docs.prebid.org/dev-docs/modules/index.html#general-modules . Please add to general modules table with description something like 'Can be used to attach rendering logic to deals, eg MASS deals.'

**Comment by massadmin** [comment]
@patmmccann In the latest PR in the doc repo we added a section in the general-modules page.
See here: [](https://github.com/prebid/prebid.github.io/pull/2807)

**Review by Fawke — CHANGES_REQUESTED** [review]
Hi @cciocov,

This change looks good. But you need to add more unit tests to achieve at least 80% coverage.

To check your coverage, please run these commands:

`gulp test-coverage` followed by `gulp view-coverage` and then navigate to your folder location.

**Comment by Fawke** [comment]
Adding @FilipStamenkovic for a second review since he already reviewed this previously. If busy, please feel free to re-assign.

**Comment by cciocov** [comment]
Hi @Fawke 

Just added more tests, thanks.

File: test/spec/modules/mass_spec.js [hunk 1]
```diff
@@ -113,4 +117,35 @@ describe('MASS Module', function() {
...
+
+  it('should match bids by deal ID with the default matcher', function() {
+    const match = useDefaultMatch(/abc/);
+
+    expect(match({dealId: 'abc'})).to.equal(true);
+    expect(match({dealId: 'xyz'})).to.equal(false);
+  });
+
+  it('should have a default renderer', function() {
+    const render = useDefaultRender('http://example.com/render.js', 'abc');
```
Comment (patmmccann) [inline test/spec/modules/mass_spec.js:145]: Can you make this secure, and in the above place as well? Even though it isn't used,  we occasionally grep the repo for nonsecure urls to ensure secure everywhere compliance

Comment (cciocov) [reply test/spec/modules/mass_spec.js]: @patmmccann it's done

**Review by FilipStamenkovic — APPROVED** [review]
LGTM.  
But I would still like to see working example in integration pages, with some bidder with test params, where bidder is bidding with appropriate `mass` flags and then this module does its magic.  

If not possible in this PR, then in some upcoming PR it would be much appreciated.

**Comment by massadmin** [comment]
@FilipStamenkovic Thank you! Yes, we will work with the respective providers to have hopefully something arranged.

**Review by Fawke — APPROVED** [review]
LGTM

**Pull request merged**